droidmediabuffer.cpp

```diff
@@ -138,31 +140,73 @@ DroidMediaBuffer *droid_media_buffer_create_from_raw_data(uint32_t w, uint32_t h
     return NULL;
   }
 
-  dstStride = buffer->getStride();
+  // Common
+  linesY = h;
+  lineWidthY = w;
+  dstStrideY = buffer->getStride();
+
+  switch (format) {
+  case HAL_PIXEL_FORMAT_YV12:
+    dstStrideUV = dstStrideY / 2;
+    // U plane is followed by V plane. So let's do 2-in-1 by not halving h.
+    linesUV = h;
+    lineWidthUV = w / 2;
+    break;
+  case HAL_PIXEL_FORMAT_YCrCb_420_SP:
+  case /* QOMX_COLOR_FormatYUV420PackedSemiPlanar32m */ 0x7FA30C04:
+  case /* QOMX_COLOR_FormatYUV420PackedSemiPlanar64x32Tile2m8ka */ 0x7FA30C03:
+    linesUV = h / 2;
+    lineWidthUV = w;
+    dstStrideUV = dstStrideY;
+    break;
+  case HAL_PIXEL_FORMAT_YCbCr_422_SP:
+    linesUV = h; // Twice the lines as 4:2:0.
+    lineWidthUV = w;
+    dstStrideUV = dstStrideY;
+    break;
+  case HAL_PIXEL_FORMAT_YCbCr_422_I:
+    // Not sure how this format works. However, it seems like the size is the same.
+    // Let's try this. If someone encounter this and know how it works, the logging
+    // line will squarely point itself out.
+
+    linesUV = h;
+    lineWidthUV = w;
+    dstStrideUV = dstStrideY;
+    ALOGW("Encountered HAL_PIXEL_FORMAT_YCbCr_422_I. Please check if these values make sense."
+          "(format = %d, dstStrideUV = %d, strideUV = %d)", format, dstStrideUV, strideUV);
```

Comment:
Missing break. And I generally don't like adding "I'm not sure how this is supposed to work" kind of code.

Reply:
Could you please point out where I can find out the correct values for this format?

Reply:
I've fixed that now. Please take another look.